olive/engine/engine.py

```diff
@@ -202,37 +224,231 @@ def run(
+
+        iter_num = 1
+
+        # get the next step
+        next_step = self.search_strategy.next_step()
+        assert next_step is not None, "Search strategy returned None for the first step"
+
+        # get the model id of the first input model
+        model_id = next_step["model_id"]
+        if model_id == input_model_id:
+            model = input_model
+        else:
+            model = self._load_model(model_id)
+
+        if verbose:
+            logger.info(f"Step {iter_num} with search point {next_step['search_point']} ...")
+
+        # run all the passes in the step
+        (
+            _,
+            signal,
+            model_ids,
+        ) = self.run_passes(next_step["passes"], model, model_id, accelerator_spec, verbose)
+        model_id = model_ids[-1]
+
+        prefix_output_name = f"{output_name}_" if output_name is not None else ""
+        # save the model to output_dir
+        output_model_name = f"{prefix_output_name}model"
+        output_model_json = cache_utils.save_model(model_id, output_dir, output_model_name, self._config.cache_dir)
+
+        # save the evaluation results to output_dir
+        result_name = f"{prefix_output_name}metrics"
+        results_path = output_dir / f"{result_name}.json"
+        if signal is not None:
+            json.dump(signal, open(results_path, "w"), indent=4)
+
+        output = {"model": output_model_json}
+        if signal is not None:
+            output["metrics"] = signal
+
+        # Package output model as artifacts if no search
+        if packaging_config:
+            logger.info("Package output model as artifacts")
+            generate_output_artifacts(
+                packaging_config,
+                self.footprints[accelerator_spec],
+                self.footprints[accelerator_spec].get_last_node(),
+                output_dir,
+            )
+
+        return output
+
+    def run_one_accelerator(
+        self,
+        input_model: OliveModel,
+        input_model_id: str,
+        accelerator_spec,
+        packaging_config: Optional[PackagingConfig] = None,
+        verbose: bool = False,
+        output_dir: str = None,
+        output_name: str = None,
+    ):
+        """
+        Run all the registered Olive passes on the input model and produce one or more candidate models.
+
+        if search strategy is None, all passes are run in the order they were registered.
+        Save the final model to {output_dir}/{output_name}_model and json file to {output_dir}/{output_name}_model.json
+        Save evaluation results of the final model, if any, to {output_dir}/{output_name}_metrics.json
+        Return {"model": final_model_json, "metrics": evaluation_results}
+
+        if search strategy is not None, run the search strategy to find candidate models.
+        TODO: save the results using updated RunResult
+        """
+
+        prefix_output_name = f"{output_name}_" if output_name is not None else ""
```

Comment:
We may need to add `acelerator_spec` into `prefix_output_name`. Otherwise, each accelerator_spec will write the same file for output.

Reply:
will do